lib/ii.c

```diff
@@ -10217,7 +10231,7 @@ token_info_build_near_phrase(grn_ctx *ctx,
     }
 
     uint32_t n_before = data->n_token_infos;
-    rc = token_info_build_phrase(ctx, data, phrase, phrase_len);
+    rc = token_info_build_phrase(ctx, data, phrase, phrase_len, GRN_TOKENIZE_ADD);
```

Comment:
We should not use `GRN_TOKENIZE_ADD` in search. It may add new tokens.

Reply:
Hmm... I see.

A current problem is that when a type of tokenizer is N-gram, `n_token_infos` seems be not expected value.

The `interval` for `near` series are calculated as:

`interval` = `interval between tops of two words` - `n_token_infos of the left word` + 1.

For example, thinking about `abcdefg`.
And if `* NP "abc ef"` is executed:

`interval between tops of two words`: interval between a and e = 4
`n_token_infos of the left word`: token num of `abc` = num of `ab bc` by Bi-gram and `GET_TOKENIZE_GET` = 2

as a result, `interval` is 3.

But...  `abcdefg` is normalized as `ab bc cd de ef fg` by Bi-gram.
I think `abc` should be considered as having token `cd` for `interval` calculation because it contains part of `abc` (`c`), but `abc` is normalized `ab bc` by Bi-gram and `GET_TOKENIZE_GET` .
The reason why I use `GET_TOKENIZE_ADD` is because when `GET_TOKENIZE_ADD` specified, `abc` is normalized as `ab bc c` and the number of token matches expected. It's just that as a result, it doesn't seem to be semantically correct though...

Is there the way to tokenize `abc` to `ab bc c` by Bi-gram and `GET_TOKENIZE_GET`?
I would like to get suffix tokens shorter than N-gram's N (e.g. N = 2 when Bi-gram, and N = 3 when Tri-gram).

Reply:
Or, do you have any other good idea for calculating `interval` ?
I can't think of any good ideas...

Reply:
We should not use `c` for search because it's redundant. It causes extra needless search. It has a performance penalty.

`grn_token_have_overlap()` may help us...